src/libponyrt/actor/actor.c

```diff
@@ -866,7 +879,10 @@ void* pony_alloc_large_final(pony_ctx_t* ctx, size_t size)
 PONY_API void pony_triggergc(pony_ctx_t* ctx)
 {
   pony_assert(ctx->current != NULL);
-  ctx->current->heap.next_gc = 0;
+
+  // only trigger gc if actor allocated something on the heap
+  if(ctx->current->heap.used > 0)
+    ctx->current->heap.next_gc = 0;
```

Comment:
This isn't technically needed as the logic that decides if we gc or not in `ponyint_heap_startgc` is:

```c
  if(heap->used <= heap->next_gc)
    return false;
```

so setting next_gc to 0 when heap.used is 0 will not result in gc happening.

I thin this should be removed as it is extra cycles being used.

Reply:
Is the logic that you really wanted here "only gc if we have allocated anything since the last time"? I think that is what you want.

Reply:
actually i see what you are doing here. if we did any memory usage at all, even 1 byte, then try gc'ing.

but, isn't 0 still something that could be of issue? what if we are holding a ton of stuff that could be freed but didn't allocate anything else on the heap. is that possible?

Reply:
Your first comment was correct. I forgot about the same check already existed in `ponyint_heap_startgc`.